### PR TITLE
ChannelList refactor for Integration Selector / Browse Channels

### PR DESCRIPTION
**Pull request by javaguirre**
#### Summary

Move ChannelList to components, refactor Integration Selector to use it and make it selectable as the UserList.

This is the trickiest DataSource to implement and refactor, the next one will be new, so we won't need to take care of backwards compatibility.

The ChannelList is now "selectable", so It gets the CompassIcon for selections when is `selectable`, off y default.

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note
Use channelList as a component in browseChannels and integrationSelector
```

### CODE REVIEW

File: app/components/channel_list_row/index.tsx [hunk 1]
```diff
@@ -19,13 +19,17 @@ type Props = {
...
     selectable?: boolean;
     selected?: boolean;
+    disabled?: boolean;
 }
 
 const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
     return {
         titleContainer: {
             marginLeft: 16,
+            paddingHorizontal: 10,
```
Comment (javaguirre) [inline app/components/channel_list_row/index.tsx:29]: This change was needed to make it "selectable", let me know if you have any concern, I've checked in browse channels and the integration selector and the design is good.

File: app/components/channel_list_row/index.tsx [hunk 2]
```diff
@@ -68,7 +80,8 @@ export default function ChannelListRow({
             return null;
         }
 
-        const color = selected ? theme.buttonBg : theme.centerChannelColor;
```
Comment (javaguirre) [inline app/components/channel_list_row/index.tsx:71]: Following UserList on this one.

File: app/components/user_list/index.tsx [hunk 3]
```diff
@@ -127,6 +129,8 @@ export default function UserList({
     term,
     testID,
     tutorialWatched,
+    selectable = false,
+    disabled = false,
```
Comment (javaguirre) [inline app/components/user_list/index.tsx:133]: Setting disable so CreateDirectMessage can continue working as it was.

File: app/components/user_list/index.tsx [hunk 4]
```diff
@@ -175,8 +178,8 @@ export default function UserList({
                 isMyUser={currentUserId === item.id}
                 onPress={handleSelectProfile}
                 onLongPress={openUserProfile}
-                disabled={!canAdd}
-                selectable={true}
+                disabled={disabled}
```
Comment (javaguirre) [inline app/components/user_list/index.tsx:181]: I forgot to change this on the UserList refactor, it works better now. The selector CompassIcon were being showed when it was not multiselect, that's now fixed.

**Comment by larkox** [comment]
@javaguirre Unless you find it urgent, I would postpone this review (and therefore the PR) until we cut the release candidate, so this change goes into 2.1.

What do you think? Or do you see any strong reason to have it done for 2.0?

**Comment by javaguirre** [comment]
> @javaguirre Unless you find it urgent, I would postpone this review (and therefore the PR) until we cut the release candidate, so this change goes into 2.1.
> 
> What do you think? Or do you see any strong reason to have it done for 2.0?

I think that's fine. :-)

**Comment by enahum** [comment]
This PR has conflicts that need resolution

**Comment by javaguirre** [comment]
@larkox should we revisit this?

File: app/components/channel_list_row/index.tsx (hunk 1 above)
Comment (larkox) [reply app/components/channel_list_row/index.tsx:29]: @matthewbirtch Can you sanity check this?

**Review by larkox — APPROVED** [review]
Codewise looks good to me.

File: app/components/channel_list_row/index.tsx (hunk 1 above)
Comment (matthewbirtch) [reply app/components/channel_list_row/index.tsx:29]: Could you share some screenshots to see the UI? Or is there a better way for me to test this?

**Comment by javaguirre** [comment]
> Codewise looks good to me.

That's great! It seems your lessons grew on me. 💯

File: app/components/channel_list_row/index.tsx (hunk 1 above)
Comment (matthewbirtch) [reply app/components/channel_list_row/index.tsx:29]: @javaguirre based on the screen capture here, I'm seeing inconsistencies with the `Find channels` modal and the `Select a channel` UI.

1. In the `Select a channel` UI, there seems to be extra space between the icon and the channel name.
2. This view uses dividers, whereas the `Find channesl` doesn't
3. The search input does not have the same margin as it does in the `Find channels` view, so it's sitting too close to the edges
4. The search input has strange highlight when active

image

Comment (javaguirre) [reply app/components/channel_list_row/index.tsx:29]: Thank you @matthewbirtch! I'll fix them when I have time.

**Comment by enahum** [comment]
closing the PR for now

**Pull request closed without merging**